crates/biome_json_syntax/src/file_source.rs

```diff
@@ -214,7 +214,9 @@ impl JsonFileSource {
             return Ok(Self::json_allow_comments());
         }
         if let Some(Component::Normal(parent_dir)) = path.components().rev().nth(1) {
-            if Self::is_well_known_json_allow_comments_directory(parent_dir) {
+            if Self::is_well_known_json_allow_comments_directory(parent_dir)
+                && file_name.as_encoded_bytes().ends_with(b".json")
```

Comment:
JSON files might also be valid? Wouldn't it be more future proof to use `try_from_extension`?

Reply:
JSONC files are already handled because of their extension? I am unsure if we want to include all the cases of `try_from_extension` because only `json` files are expected to allow comments in the `.vscode` and `.zed` directories?